python/ray/data/dataset_pipeline.py

```diff
@@ -700,6 +682,325 @@ def __iter__(self):
 
         return EpochDelimitedIterator(self, max_epoch)
 
+    def map(
+        self,
+        fn: RowUDF,
+        *,
+        compute: Union[str, ComputeStrategy] = None,
+        **ray_remote_args,
+    ) -> "DatasetPipeline[U]":
+        """Apply ``Dataset.map`` to each dataset/window in this pipeline."""
```

Comment:
This doesn't have to happen in this PR, but we should expand these docstrings to full docstrings, including API semantics + documentation for each parameter.

Reply:
Sure, let me do it in another PR as followup.